spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -317,14 +1032,9 @@ public Message<OutgoingMessage> processMessage(Message<IncomingMessage> msg) {
 }
 ----
 
-===== Implicit acknowledgement
-
-Incoming messages may be implicitly acknowledged if the incoming message is not wrapped in `Message`. When it is acknowledged depends on the shape of the method. For simple method streams, the message should be acknowledged after the return value of the method is redeemed - for synchronous methods, this is when the method returns, for asynchronous methods, this is when the returned `CompletionStage` is redeemed.
-
-If the method returns a subscriber, then the message should be acknowledged before being sent to the subscriber.
-
-If the method returns a processor, then it is assumed that there is a 1:1 relationship between incoming and outgoing messages, and the implementation should acknowledge incoming messages as corresponding outgoing messages are published.
+The implementation is reponsible to enforce the acknowledgement strategy defined by the user when the `@Acknowledgement` policy is used. 
+If the annotation is not used, the default policy must be enforced.
 
-==== Custom message types
+// ==== Custom message types
 
-Implementations may provide custom subtypes of `Message` as an end user API, for the purpose of providing additional metadata associated with the messaging provider. If an application declares that it consumes or produces a subtype of the `Message` wrapper, the implementation should validate that the messaging provider supports that message wrapper type, and raise a deployment exception if it doesn't.
+// Implementations may provide custom subtypes of `Message` as an end user API, for the purpose of providing additional metadata associated with the messaging provider. If an application declares that it consumes or produces a subtype of the `Message` wrapper, the implementation should validate that the messaging provider supports that message wrapper type, and raise a deployment exception if it doesn't.
```

Comment:
Only minor comments. Apologies that it took me a number of days, I had an expected illness/death in the family and have been out of the office a lot recently.